### DISCUSSION TITLE
Questions about the kernel(MatDiffusion).

### DISCUSSION THREAD

**Discussion by wowodejiajia** [idaholab/moose#22247] [Q&A Modules: Phase field] [answered]
Dear MOOSE experts,

In this article (Zijian Hong, Venkatasubramanian Viswanathan. Phase-Field simulations of lithium dendrite growth with open-source software. ACS Energy Lett., 2018, 3, 7, 1737–1743), the author has compiled a kernel(Migration). But I think it's OK to use the original kernel(MatDiffusion). Then I tried to simulate, but the result was different from the original. Can you help me see the difference between the two kernels?

Thank you.
```
Original file input.i
[Kernels]
  [./Diffusion2]
    # Migration.
    type = Migration
    variable = w
    cv = eta
    Q_name = 0.
    QM_name = DN
    cp = pot
  [../]

I modified input.i
  [./Diffusion2]
    # Originally Migration, I changed it to MatDiffusion.
    type = MatDiffusion
    variable = pot
    diffusivity = DN
    args = 'eta w'
  [../]

[Materials]
  [./Migration_coefficient]
    # Migration coefficient.
    type = DerivativeParsedMaterial
    function = '-cl*(1-h)*AA*M0*(1-h)'
    args = 'eta w'
    f_name = DN
    derivative_order = 1
    material_property_names = 'M0 AA cl:=D[f1,w] h'
  [../]
```
[Migration.txt](https://github.com/idaholab/moose/files/9661279/Migration.txt)

**Comment by GiudGiud** [comment, marked as answer]
Hello

The residual term is different. 
Your input, roughly:
`grad_v[_qp] * _grad_test[_i][_qp] * -cl*(1-h)*AA*M0*(1-h)`
Theirs:
`_Q[_qp]*_grad_cv[_qp]*_grad_test[_i][_qp]+_QM[_qp]*_grad_cp[_qp]*_grad_test[_i][_qp];`
It seems you are missing a second variable

I'd encourage you to output the equivalent of the diffusion coefficient in the original simulation and see if you can reproduce that with your material. 

Guillaume

**Reply by wowodejiajia** [reply]
Hello

What is the function of the v in the kernel（MatDiffusion）? I read the instructions on the website, but I didn't understand  where v works. Can you give me an example?

Thank you.

**Reply by GiudGiud** [reply]
It s to couple another variable than the variable attached to the equation (usually the one featured in the time derivative)
